src/main/java/org/osjava/sj/loader/convert/BeanConverter.java

```diff
@@ -57,6 +73,192 @@
 
     private static Logger LOGGER = LoggerFactory.getLogger(BeanConverter.class);
 
+    private static final Set<String> trueValues;
+    private static final Set<String> falseValues;
+
+    public static final int OLD_ENUM_STYLE = Modifier.FINAL | Modifier.PUBLIC | Modifier.STATIC;
+
+    private static final List<SimpleDateFormat> dateTimeFormats;
+    private static final List<SimpleDateFormat> dateFormats;
+    private static final List<SimpleDateFormat> timeFormats;
+
+    static {
+        trueValues = new TreeSet<String>(String.CASE_INSENSITIVE_ORDER);
+        falseValues = new TreeSet<String>(String.CASE_INSENSITIVE_ORDER);
+        Collections.addAll(trueValues, "true", "t", "yes", "y", "on", "1", "x", "-1");
+        Collections.addAll(falseValues, "false", "f", "no", "n", "off", "0", "");
+        dateTimeFormats = new ArrayList<SimpleDateFormat>();
+        Collections.addAll(dateTimeFormats, new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ"),
+            new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSXXX"),
+            new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss"),
+            new SimpleDateFormat("yyyy-MM-dd'T'HH:mm"),
+            new SimpleDateFormat("yyyy-MM-dd"));
+        for(SimpleDateFormat sdf : dateTimeFormats) {
+            sdf.setLenient(false);
+        }
+        dateFormats = new ArrayList<SimpleDateFormat>();
+        Collections.addAll(dateFormats, new SimpleDateFormat("yyyy-MM-dd"));
+        for(SimpleDateFormat sdf : dateFormats) {
+            sdf.setLenient(false);
+        }
+        timeFormats = new ArrayList<SimpleDateFormat>();
+        Collections.addAll(timeFormats, new SimpleDateFormat("HH:mm:ss.SSSZ"),
+            new SimpleDateFormat("HH:mm:ss.SSSXXX"),
+            new SimpleDateFormat("HH:mm:ss"),
+            new SimpleDateFormat("HH:mm"));
+        for(SimpleDateFormat sdf : timeFormats) {
+            sdf.setLenient(false);
+        }
+    }
+
+    private static boolean convertStringToBooleanPrimitive(String value) {
+        if(trueValues.contains(value)) {
+            return true;
+        }
+        if(falseValues.contains(value)) {
+            return false;
+        }
+        throw new RuntimeException("The value, \"" + value + "\", can not be converted to a boolean.");
```

Comment:
Please support properties without a value, e. g.

`booleanPrimitive = <--- value omitted`

At the moment it results in

`java.lang.RuntimeException: The value, "", could not be converted to a "boolean".
	at org.osjava.sj.loader.convert.BeanConverter.convert(BeanConverter.java:164)`

I would propose this strategy:

Omitted properties results in null, false or 0, but not "", dependent on the field type. There should be nothing to do as it is Java's default behaviour when instantiating an object.
Properties without a value results in "", false or 0 dependent on the field type. In these cases the property's value in the Property-Object is always an empty String.

Reply:
On 3. Dec 2017, at 03:42, jwaterloo notifications@github.com wrote:

As far as "" for boolean, I can't say. Unconfigured properties take the
default value set up by whoever created the class. Since configuring is
explicitly setting the value than the configurer could be expected to
use a reasonably correct value. Note it is supported for Boolean as it
could be understood for null.